### PR TITLE
Documentation work; added platform-specific notes for nix

### PR DESCRIPTION
**Pull request by pelasgus**
…v env via nix for rust dev.

### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Docs
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### Checklist
- [ ] This PR will resolve #___
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary
- [ ] It can be built on all targets and pass CI/CD.

### CODE REVIEW

File: README.md
Comment (amrbashir) [inline README.md]: kepp the original header

File: README.md
Comment (amrbashir) [inline README.md]:
```suggestion
#### NixOS
```

File: README.md
Comment (amrbashir) [inline README.md]: We should only mention the packages related to this repo, other packages are for user preference:
```suggestion
  packages = with pkgs; [
    pkg-config
    webkitgtk_4_1
  ];
```

File: README.md
Comment (amrbashir) [inline README.md]: is there a reason to recommend the unstable channel instead of stable?

File: README.md
Comment (amrbashir) [inline README.md]: Please remove this, the repo is not a place to document what Nix or NixOS is, if you need nix instructions then you already know what nix is.

File: README.md
Comment (amrbashir) [inline README.md]: Please move this section to last after Fedora instructions

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: README.md
Comment (pelasgus) [reply README.md]: unstable channel on nix is essentially rolling release, thus in my subjective opinion, more suitable for development purposes.

File: README.md
Comment (pelasgus) [reply README.md]: Reverted to original

**Comment by pelasgus** [comment]
Implemented all requested changes; pending approval.

File: README.md
Comment (amrbashir) [inline README.md:111]: does nix have `libappindicator` or `libayatana` pkgs? if so, we should add them here with a note like other OSes

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Oh looks like you didn't sign your first commit and I can't merge the PR unless all commits are signed. You need to setup [commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits), then you can sign past commit like [this](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits/1123928#1123928) for example.

**Comment by pelasgus** [comment]
@amrbashir just merge this anyway, your commits cannot be fully verified, its you and my original commit that are blocking merge. Please rectify, cheers

**Comment by amrbashir** [comment]
Partially verified commits are fine, but you have one commit at the beginning of the PR that is not verified. 
image

You can try squashing all commits into one verified commit or close this PR and re-open a new one. 

Otherwise our guidelines and repo configuration prevents anyone from bypassing this requirement.

**Pull request closed without merging**
